platform/darwin/src/MGLNetworkConfiguration.m

```diff
@@ -65,9 +65,13 @@ - (NSURLSessionConfiguration *)defaultSessionConfiguration {
 }
 
 - (void)startDownloadEvent:(NSString *)urlString type:(NSString *)resourceType {
-    if (urlString && ![self.events objectForKey:urlString]) {
-        [self.events setObject:@{ MGLStartTime: [NSDate date], MGLResourceType: resourceType } forKey:urlString];
+    NSString *retainedUrlString = urlString;
+    NSString *retainedResourceType = resourceType;
+    if (retainedUrlString && !self.events[retainedUrlString]) {
+        self.events[retainedUrlString] = @{ MGLStartTime: [NSDate date], MGLResourceType: retainedResourceType };
```

Comment:
I think we need to change the events property to be atomic:

`@property (atomic, strong) NSMutableDictionary<NSString*, NSDictionary*> *events;`

since reading/writing/deleting could happen at the same time.

Reply:
This property gets called multiple times. I consider the atomicity but in practice found no reason to make this an atomic property when the crash is raised for a different reason.